src/realm/alloc.hpp

```diff
@@ -310,7 +310,8 @@ class Allocator {
     friend class Group;
     friend class WrappedAllocator;
     friend class Obj;
-    friend class CollectionBase;
+    template <class>
```

Comment:
Woah, Allocator is a popular class! Look at all its friends! Is this because alloc.hpp is a public header?

Reply:
Good point, might be worth taking a look at it.